### ISSUE TITLE
Select by default with Goto

### ISSUE THREAD

**Issue by schecko** [helix-editor/helix#5127]
By default, basic movements such as "w" or "b" will highlight in the wake of the movement, rather than requiring something like "vw" to select before performing an action. Should more movements have this feature? In particular, the "g" commands could benefit from this. For example, to change to the end of the line would be "vglc", the same functionality in vim would be "ce". With this change, "vglc" becomes just "glc".

**Comment by archseer** [comment]
This is not a new proposal: https://github.com/helix-editor/helix/issues/1630

The question is, there's `gl`/`ge` etc that are movements, but then `gd`/`gi` etc are jumps. Do we diverge and have some movements select, and others jump?

**Comment by schecko** [comment]
Unless the jump is to another file, i think its fine to just make everything select by default? I can see it having its place for something like `gt` or `gc` etc.

**Comment by CptPotato** [comment]
This might be a bit controversial, but I actually prefer the goto bindings to not select. In my mental model I consider all of the current goto movements "jumps" rather than "relative movements".
With the selection, in the best case this happens to select what I wanted so I save a `v` press. In the worst case it selects something I didn't want, so I have to cancel the automatic selection and reselect what I actually wanted. For example deleting line 123 would turn into `123gg;Xd`.

I guess it's probably down to personal preference and how you use the editor (i.e. if the default ends up being what you want in most cases). I usually don't want to select from the starting position when using any of the goto commands. A common one I use is `gli` to insert before the tailing `;`/`,`/`.` on a line. Or knowing that there's an error on line 200, `200ggi` will go there in insert mode.

However, in case selection is implemented for goto commands, I could probably get around this by remapping the key binds to explicitly deselecting afterwards.

**Comment by pascalkuthe** [comment]
I think having separate versions of the goto commands that select might be reasonable/uncontroversial addition so users could change their keymap as they desire.

I used to want this feature for `gl` and `gh` but actually found that I prefer `t<ret>` and `T<ret>` for selecting to line end/start. It doesn't require me to remember an extra keybindings altough it's slightly harder to reach. Other goto commands are probably used rarely enough that it's not a big deal to use `v` to select if that is what you want.

**Comment by sicher** [comment]
I don't know if this is a bug, but I get the same result with T<ret> as with F<ret>. In both cases, the selection includes the previous return character:
![image](https://github.com/helix-editor/helix/assets/65613/ee6e0848-e59a-4fde-84cf-656e027ab443)

**Comment by pascalkuthe** [comment]
that seems like a bug, most likely I would guess that you  are using windows/a file with crfl line ending and `T<ret>`  doesn't handle that correctly

**Comment by sicher** [comment]
Yes, I only see that on windows. I'll report that as a separate issue when I'm back on my work machine.
